### PR TITLE
feat: user and authentification

### PR DESCRIPTION
**Pull request by justinDev91**
# Ticket Notion

# Screen (si besoin)

# Autre information

### CODE REVIEW

**Review by PatRider64 — APPROVED** [review]

File: src/src/Form/UserEditType.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            ->add('email');
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'data_class' => User::class,
+        ]);
+    }
+}
```
Comment (MisterAzix) [inline src/src/Form/UserEditType.php:26]: Why a second form builder ? You can just make a condition to check if you are in creation or editing, ask me if u want to see how to do it ;)

File: src/src/Entity/User.php [hunk 2]
```diff
@@ -3,14 +3,14 @@
...
-use Doctrine\Common\Collections\ArrayCollection;
-use Doctrine\Common\Collections\Collection;
 use Doctrine\DBAL\Types\Types;
 use Doctrine\ORM\Mapping as ORM;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
 use Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface;
 use Symfony\Component\Security\Core\User\UserInterface;
 
 #[ORM\Entity(repositoryClass: UserRepository::class)]
+#[UniqueEntity(fields: ['email'], message: 'There is already an account with this email')]
```
Comment (MisterAzix) [inline src/src/Entity/User.php:13]: Nice 👌

File: src/src/Form/UserType.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+class UserType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('name')
+            ->add('first_name')
+            ->add('email')
+            ->add('password');
+    }
```
Comment (MisterAzix) [inline src/src/Form/UserType.php:20]:
```suggestion
    public function buildForm(FormBuilderInterface $builder, array $options): void
    {
        $user = $options['data'] ?? null;
        $isEdit = $user && $user->getId();
    
        $builder
            ->add('name')
            ->add('first_name')
            ->add('email');
            
        if ($isEdit) {
          $builder->add('password');
        }
    }
```

File: src/src/Controller/UserController.php [hunk 4]
```diff
@@ -3,96 +3,90 @@
...
-    #[Route('/user/{email}/modify', name: 'app_user_modify')]
-    public function modify(User $user): Response
+    #[Route('/{id}/edit', name: 'app_user_edit', methods: ['GET', 'POST'])]
+    public function edit(Request $request, User $user, UserRepository $userRepository): Response
     {
-        return $this->render('user/user_modify.html.twig', [
-            'user' => $user
-        ]);
-    }
+        $form = $this->createForm(UserEditType::class, $user);
```
Comment (MisterAzix) [inline src/src/Controller/UserController.php:68]:
```suggestion
        $form = $this->createForm(UserType::class, $user);
```

**Comment by Mowdyy** [comment]
Pas mal pas mal, mais manque de personnalité

File: src/src/Form/UserEditType.php (hunk 1 above)
Comment (justinDev91) [reply src/src/Form/UserEditType.php:26]: Thx pepito haha

**Review by MisterAzix — APPROVED** [review]

**Review by Triips-TheCoder — APPROVED** [review]

**Review by PatRider64 — APPROVED** [review]

**Comment by justinDev91** [comment]
> Pas mal pas mal, mais manque de personnalité

Très fort @Mowdyy

**Pull request merged**